core/src/main/java/org/apache/calcite/rel/RelNode.java

```diff
@@ -246,6 +246,8 @@ default String getDigest() {
     return getRelDigest().toString();
   }
 
+  String getTreeDigest();
```

Comment:
The new method should have a javadoc

Reply:
Revised, really sorry for forgetting this.